app/hooks/use-display-currency.ts

```diff
@@ -32,54 +31,67 @@ gql`
   }
 `
 
-const defaultDisplayCurrency = {
+const usdDisplayCurrency = {
   symbol: "$",
   id: "USD",
   fractionDigits: 2,
 }
 
+const defaultDisplayCurrency = usdDisplayCurrency
+
+const formatCurrencyHelper = ({
+  amountInMajorUnits,
+  symbol,
+  fractionDigits,
+  withSign = true,
+}: {
+  amountInMajorUnits: number | string
+  symbol: string
+  fractionDigits: number
+  withSign?: boolean
+}) => {
+  const isNegative = Number(amountInMajorUnits) < 0
+  const amountStr = Intl.NumberFormat("en-US", {
+    minimumFractionDigits: fractionDigits,
+    maximumFractionDigits: fractionDigits,
+    // FIXME this workaround of using .format and not .formatNumber is
+    // because hermes haven't fully implemented Intl.NumberFormat yet
+  }).format(Math.abs(Number(amountInMajorUnits)))
+  return `${isNegative && withSign ? "-" : ""}${symbol}${amountStr}`
+}
+
 export const useDisplayCurrency = () => {
   const isAuthed = useIsAuthed()
   const { data: dataCurrencyList } = useCurrencyListQuery({ skip: !isAuthed })
-
   const { data } = useRealtimePriceQuery({ skip: !isAuthed })
 
   const displayCurrency =
     data?.me?.defaultAccount?.realtimePrice?.denominatorCurrency ||
     defaultDisplayCurrency.id
 
-  const displayCurrencyInfo = useMemo(() => {
+  const displayCurrencyDictionary = useMemo(() => {
     const currencyList = dataCurrencyList?.currencyList || []
-    return (
-      currencyList.find((currency) => currency.id === displayCurrency) ||
-      defaultDisplayCurrency
-    )
-  }, [dataCurrencyList, displayCurrency])
+    return currencyList.reduce((acc, currency) => {
```

Comment:
why using a reducer here / creating a dictionnary? There'll be at most 150 currencies, the loop should run super quickly. not sure it's worth the code complexity to create mapping imo. useMemo already provide some optimization.

Reply:
With tx history you need to be able to display any currency not just the current display currency, so without the dictionary you need to loop through the list for every amount displayed on every render. It still may not be worth the optimization, but we have optimized for less in the past.

Reply:
oh right ok. 

we're still doing N * M, N been transaction list and M been currency list. so that make more sense. didn't had the N in mind.